.github/workflows/test.yml

```
@@ -0,0 +1,84 @@
+---
+# Source: https://github.com/hendrikmaus/rust-workflows
+#
+# Continuous Integration Workflow For Rust
+name: CI
+
+# Define the triggers; usually merges to the repository
+# default branch and pull-requests
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - ".vscode/**"
+      - "**.md"
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+    paths-ignore:
+      - ".vscode/**"
+      - "**.md"
+
+env:
+  CARGO_TERM_COLOR: always
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  fmt:
+    name: Format
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
+      - uses: actions-rs/toolchain@568dc894a7f9e32ffd9bb7d7a6cebb784cdaa2b0
+        with:
+          toolchain: stable
+          components: rustfmt
+          profile: minimal
+      - uses: actions-rs/cargo@4ff6ec2846f6e7217c1a9b0b503506665f134c4b
+        with:
+          command: fmt
+          args: --all -- --check
+
+  test:
+    name: Test
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
+    runs-on: ${{matrix.os}}
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
+      - uses: actions-rs/toolchain@568dc894a7f9e32ffd9bb7d7a6cebb784cdaa2b0
+        with:
+          toolchain: stable
+      - uses: Swatinem/rust-cache@1232abb8968faf344409165de17cbf9e7f340fd8
+      - uses: actions-rs/cargo@4ff6ec2846f6e7217c1a9b0b503506665f134c4b
+        with:
+          command: build
+      - uses: actions-rs/cargo@4ff6ec2846f6e7217c1a9b0b503506665f134c4b
+        with:
+          command: test
+
+  clippy:
+    name: Clippy Check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
+      - uses: actions-rs/toolchain@568dc894a7f9e32ffd9bb7d7a6cebb784cdaa2b0
+        with:
+          toolchain: stable
+          profile: minimal
+      - run: rustup component add clippy
+      - uses: Swatinem/rust-cache@1232abb8968faf344409165de17cbf9e7f340fd8
+      - uses: actions-rs/clippy-check@9d09632661e31982c0be8af59aeb96b680641bc4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --all-features
```

```suggestion
          args: --all-features
          name: CI / Clippy run
```
This should make for a nicer name to appear in the "checks" section of a pull-request. By default, the job's name appears as `clippy_check`.

---

The name currently appears as `CI / Clippy Check`, the name is configured here

https://github.com/ObiWanRohan/Fornjot/blob/6155a28d9ee469440edb479134b9a87c1396ba5f/.github/workflows/test.yml#L71

---

The job emits a check and the step it runs emits another. I don't know a way to solve that right now.

---

I don't think it's much of a problem. Just mentioned it because it seemed related to the discussion here regarding the name displayed under "checks".